contractcourt/channel_arbitrator.go

```diff
@@ -1768,8 +1768,12 @@ func (c *ChannelArbitrator) checkCommitChainActions(height uint32,
 			continue
 		}
 
-		toChain := c.shouldGoOnChain(htlc, c.cfg.IncomingBroadcastDelta,
-			height,
+		// TODO(yy): we shoud not go on chain if there's no
+		// corresponding outgoing HTLC, which means we are the final
+		// hop. Instead, we should cancel the incoming HTLC and fail
+		// the invoice.
```

Comment:
I was thinking it'd be easier to fail the invoice than go onchain, iff we haven't sent `update_fulfill_htlc`, to avoid a force close. Removed the TODO for now as I don't think it's feasible atm.